parity-crypto/src/publickey/secret_key.rs

```diff
@@ -52,21 +53,23 @@ impl fmt::Display for Secret {
 
 impl Secret {
 	/// Creates a `Secret` from the given slice, returning `None` if the slice length != 32.
+	/// Caller is responsible to zeroize input slice.
 	pub fn copy_from_slice(key: &[u8]) -> Option<Self> {
 		if key.len() != 32 {
 			return None;
 		}
 		let mut h = H256::zero();
 		h.as_bytes_mut().copy_from_slice(&key[0..32]);
-		Some(Secret { inner: h })
+		Some(Secret { inner: Pin::new(Box::new(h)) })
```

Comment:
we can also use `Box::pin` https://doc.rust-lang.org/std/boxed/struct.Box.html#method.pin